src/Features/Core/Portable/UseConditionalExpression/ForReturn/AbstractUseConditionalExpressionForReturnDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeAnalysis.UseConditionalExpression
+{
+    internal abstract class AbstractUseConditionalExpressionForReturnDiagnosticAnalyzer<
+        TIfStatementSyntax>
+        : AbstractCodeStyleDiagnosticAnalyzer
+        where TIfStatementSyntax : SyntaxNode
+    {
+        public override bool OpenFileOnly(Workspace workspace) => false;
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory() 
+            => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
+
+        protected AbstractUseConditionalExpressionForReturnDiagnosticAnalyzer(
+            LocalizableResourceString message)
+            : base(IDEDiagnosticIds.UseConditionalExpressionForReturnDiagnosticId,
+                   new LocalizableResourceString(nameof(FeaturesResources.Convert_to_conditional_expression), FeaturesResources.ResourceManager, typeof(FeaturesResources)),
+                   message)
+        {
+        }
+         
+        protected abstract ISyntaxFactsService GetSyntaxFactsService();
+
+        protected override void InitializeWorker(AnalysisContext context)
+            => context.RegisterOperationAction(AnalyzeOperation, OperationKind.Conditional);
+
+        private void AnalyzeOperation(OperationAnalysisContext context)
```

Comment:
>AnalyzeOperation [](start = 21, length = 16)

If I'm not mistaken, this method completely coincides with the same method in AbstractUseConditionalExpressionForAssignmentDiagnosticAnalyzer. Should not we create a base class for both them sharing the method?

Reply:
i'll take a looksie.

Reply:
Done.  Great suggestion!